functions.php

```diff
@@ -53,6 +53,10 @@ function as_schedule_single_action( $timestamp, $hook, $args = array(), $group =
  * @return int The action ID.
  */
 function as_schedule_recurring_action( $timestamp, $interval_in_seconds, $hook, $args = array(), $group = '', $unique = false ) {
+	// if you pass a string by mistake, it causes a flood of scheduled actions, recurring almost every minute or less
+	if ( ! is_int( $interval_in_seconds ) ) {
```

Comment:
For backwards-compatibility reasons, a less strict approach may be preferable here.

Strings and floats should probably still be allowed, so long as they are used to convey an integer. Examples of this might include `(string) '60'` or `(float) 60.0`.

Possible iteration:

```suggestion
	$interval = (int) $interval_in_seconds;

	// We expect an integer and allow it to be passed using float and string types, but otherwise
	// should reject unexpected values.
	if ( ! is_numeric( $interval_in_seconds ) || $interval_in_seconds != $interval ) {
```

Examples of allowed values, using the above code:

- `(int) 60`
- `(float) 60.0`
- `(string) 60`

Examples of values that will be rejected:

- `(float) 60.5`
- `(string) 'twicedaily'`

Of course, if we follow this approach we would also ideally update the call to `ActionScheduler::factory()->recurring_unique()` a few lines down (and pass `$interval` rather than `$interval_in_seconds`).

Reply:
To explain; if we were introducing a new function then we could be much stricter—but it's entirely possible there is code out there that passes in a int contained in a string, or similar, and we would not wish to inadvertently break it.